integrations/database/mariadb/kubearchive.sql

```diff
@@ -55,11 +70,44 @@ CREATE TABLE `resource` (
 /*!50003 SET collation_connection  = utf8mb3_general_ci */ ;
 /*!50003 SET @saved_sql_mode       = @@sql_mode */ ;
 /*!50003 SET sql_mode              = 'STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION' */ ;
+/*!50032 DROP TRIGGER IF EXISTS set_owners */;
+DELIMITER ;;
+/*!50003 CREATE*/ /*!50017 DEFINER=`kubearchive`@`%`*/ /*!50003 TRIGGER set_owners
+  AFTER INSERT ON resource FOR EACH ROW
+    BEGIN
+      DECLARE ref, refs JSON;
+      DECLARE i, len INT DEFAULT 0;
+      DECLARE owner char(36);
+
+      SET refs = JSON_QUERY(NEW.data, '$.metadata.ownerReferences');
+      SET len = JSON_LENGTH(refs);
+      WHILE i < len DO
+        SET ref = JSON_EXTRACT(refs, CONCAT('$[', i, ']'));
+        SET owner = JSON_VALUE(ref, '$.uid');
+        IF owner IS NOT NULL THEN
+          INSERT INTO owner (uuid, owner_uuid) VALUES (NEW.uuid, owner) ON DUPLICATE KEY UPDATE uuid=uuid;
+        END IF;
+        SET i = i + 1;
+      END WHILE;
+    END */;;
```

Comment:
I am confused about this, it is commented and it doesn't seem to match the same triggers defined for postgresql

Reply:
It's different because the syntax is different. Are you saying the functionality is different?  I believe both are iterating through the `ownerReferences` of the resource and adding those records to the `owner` table.

Reply:
This is also going to change based on what we've learned with indexing.

Reply:
> It's different because the syntax is different.
I was worried about what seemed to be comments (after searching it, things like `/*!50003 CREATE*/` are conditional execution tokens that mean a minimum version of mariadb 5.00.03 is expected to run the command. 

The other thing I saw different was the name of the trigger, for mariadb is `set_owners` while in postgresql we are usign `insert_owners`. I believe we should be consistent with the names even though they would work regardless as they are internal for the database.

> This is also going to change based on what we've learned with indexing.
Right, but wanted to mention this difference as it caught my attention.